### PR TITLE
Fix typo and compiler warning

### PR DESCRIPTION
**Pull request by dschwen**
Refs #19394

### CODE REVIEW

File: framework/src/utils/Units.C [hunk 1]
```diff
@@ -428,15 +430,17 @@ MooseUnits::parse(const std::string & unit_string)
   simplify();
 }
 
-MooseUnits MooseUnits::operator*(Real f) const
+MooseUnits
+MooseUnits::operator*(Real f) const
```
Comment (lindsayad) [inline framework/src/utils/Units.C:434]:
```suggestion
MooseUnits::operator*(const Real f) const
```

File: framework/src/utils/Units.C [hunk 2]
```diff
@@ -562,7 +566,7 @@ operator<<(std::ostream & os, const MooseUnits & u)
       os << ' ';
   }
 
-  for (unsigned int i = 0; i < u._base.size(); ++i)
+  for (auto i : index_range(u._base))
```
Comment (lindsayad) [inline framework/src/utils/Units.C:569]:
```suggestion
  for (const auto i : index_range(u._base))
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
thanks for taking care of this!

**Review by lindsayad — DISMISSED** [review]
For pass by value you can label the parameter non-const in the header and const in the implementation. You may have already known that; for those methods I think it makes sense that we’ll never want to modify the parameter so I’m supportive of the change in the header.

**Comment by dschwen** [comment]
> For pass by value you can label the parameter non-const in the header and const in the implementation. You may have already known that;

I did in fact not know that! But it does make sense, since a const pass by value is an "internal matter" of the implementation and the caller shouldn't have to know about it. I was just striving for consistency between header and implementation.

**Review by lindsayad — APPROVED** [review]
Failure is unrelated. I've created an issue for this (https://github.com/idaholab/moose/issues/19960)

**Pull request merged**